util.c

```diff
@@ -89,6 +94,17 @@ warning(const char *fmt, ...)
 	}
 }
 
+
+extern void tcpdump_exit(int status)
+{
+
+#ifdef USE_LIBSMI
+    smiExit();
```

Comment:
[The smi_config man page](https://www.ibr.cs.tu-bs.de/projects/libsmi/smi_config.html) doesn't say what happens if you call `smiExit()` without having called `smiInit()`, so you might want to have a static `smiInit_called` flag in `tcpdump.c`, set it after calling `smiInit()`, and only call `smiExit()` if it's non-zero.